### ISSUE TITLE
Update for Keras 2

### ISSUE THREAD

**Issue by ErmiaAzarkhalili** [kundajelab/deeplift#26]
Hi,
Thanks for great paper and library.

I think you need to update the library for Keras 2 as the syntaxes have been changed there.

**Comment by AvantiShri** [comment]
Yup, it's on the to-do.

**Comment by ErmiaAzarkhalili** [comment]
Thanks a lot, do you know the approximated time?

**Comment by AvantiShri** [comment]
Hi Ermia, I am hesitant to give a time frame as I have a lot of other research commitments, but if you give me a sense of when you would need it by, I can use that to help me prioritize. When are you hoping to use DeepLIFT by?

**Comment by ErmiaAzarkhalili** [comment]
Hi.
Thanks.
I need to get the feature importance in autoencoder model implemented in Keras 2.

**Comment by ErmiaAzarkhalili** [comment]
I need it within next three weeks if you could make it. It would be so important as we are going to submit our work to the nature journal.

**Comment by AvantiShri** [comment]
Ok. We have not implemented DeepLIFT for autoencoders yet but you are also not the first to request it. What is your model architecture? (I am trying to ascertain if we at least support all the layers you are using)

**Comment by AvantiShri** [comment]
Also which backend (theano or tensorflow)?

**Comment by ErmiaAzarkhalili** [comment]
Thanks for your great help.

Relu, linear and sigmoid layers as well as dropout layer mostly used. I used tensorflow backend in my work.

**Comment by ghadirinia** [comment]
sorry, I did a mistake. I thought you asked as to which activation functions I used.
I implemented  the VAE with Relu, Linear and sigmoid activations.
As you know, VAE has Lambda and Merge layer and Dropout layer frequently used in such networks.

**Comment by AvantiShri** [comment]
Oh. VAEs are not deterministic. We haven't worked out DeepLIFT rules for the sampling layer of a VAE. If the VAE could be treated deterministically at test-time (that is, only relying on the encoder mean and ignoring the encoder standard deviation) then you could pretend it is a regular autoencoder and do some calculations. Are you only using the sigmoid in the output layer? Also, what kind of insight are you hoping to get from DeepLIFT?

**Comment by Ezekiel-Kruglick** [comment]
Very nice looking system, your papers are nice.

I would also love a tool that works with keras 2.x, and ideally with conv1d. I imagine many in the community would and I understand what it's like running a public repo, thanks for the work that you do!

**Comment by AvantiShri** [comment]
I'm currently working on getting it to work with Keras 2.x!

**Comment by Ezekiel-Kruglick** [comment]
Great news!

**Comment by AvantiShri** [comment]
Very sorry for the delay, I've been dealing with some health issues this past year. A keras2 compatible version is at https://github.com/kundajelab/deeplift/tree/keras2compat
